src/core/model/Model.cpp

```diff
@@ -1,14 +0,0 @@
-#include "Model.h"
-
-namespace SEMBA {
-namespace Model {
-
-Model::Model(
-	const Mesh::Unstructured& unstructuredMesh,
-	const PMGroup& physicalModels
-) : unstructuredMesh(unstructuredMesh),
-	physicalModels(physicalModels)

```

Comment:
The same as with assignment. We would have to reassign the pointers.You can check that if fails using the ctor in a test in which the original model is deleted.

Reply:
Good catch. Going to add the reassigment in the new `.hpp` file.